### ISSUE TITLE
Add missing stubs to exercises

### ISSUE THREAD

**Issue by Dark-WizZ** [exercism/java#2133]
## The problem

Right now for some exercises, a student is required to read the tests to know which functions they should implement. This is annoying, time consuming and might discourage people from doing the exercise entirely. We should provide stubs for all exercises. Stubs should provide the signature of functions and classes, just so the student is immediately ready to start writing the custom logic. This is an example of a stub, from the [Raindrops exercise](https://github.com/exercism/java/blob/main/exercises/practice/raindrops/src/main/java/RaindropConverter.java):

```java
class RaindropConverter {

    String convert(int number) {
        throw new UnsupportedOperationException("Delete this statement and write your own implementation.");
    }

}
```

All stubs must follow this format.

## Contributing to this issue

This is a big issue that will require changes to a lot of exercises. The good news is that you don't need to fix this is all exercises to contribute to this issue! PRs fixing this in just one exercise at a time or a couple of exercises at a time are more than welcome.

**Comment by andrerfcsantos** [comment]
I absolutely agree we should do this!

@Dark-WizZ I changed the description of the issue just to make the issue and what needs to be done clearer.

**Comment by andrerfcsantos** [comment]
@jcovington16 Go ahead! Are you considering changing this is all exercises or only in a subset of them?

**Comment by jcovington16** [comment]
Just a subset. After this one I'll be able to knock out some more difficult tasks. Just trying to work myself up to it.

**Comment by andrerfcsantos** [comment]
@jcovington16 You absolutely can work on this issue, sorry I haven't made that clear! What I meant to say before is that since you won't be working on all the exercises and only a subset of them, I won't mark the issue as claimed just so other people can join alongside you. it's totally ok for several people to work on this issue at the same time, ideally looking at different exercises.

**Comment by jcovington16** [comment]
Oh ok I understand now. I'll finish up the PR I have now and get started with this one before I dive into the analyzer. This one shouldn't take me long.

**Comment by jcovington16** [comment]
@Dark-WizZ @andrerfcsantos Do we know specifically which exercises need these stubs implemented? And which tracks as well?

**Comment by andrerfcsantos** [comment]
It seems here in the Java track, all "medium" difficulty exercises or above don't include stubs, so probably those are a good start, but there might be more. A quick `jq` query for exercises that are medium difficulty or above gives me this:

<details>
  <summary>jq query</summary>
  
```console
$ jq '.exercises.practice | map(select(.difficulty >= 4)) | map(.slug) | sort' config.json
[
  "affine-cipher",
  "all-your-base",
  "allergies",
  "alphametics",
  "anagram",
  "atbash-cipher",
  "bank-account",
  "beer-song",
  "binary-search",
  "binary-search-tree",
  "bob",
  "book-store",
  "bowling",
  "change",
  "circular-buffer",
  "clock",
  "collatz-conjecture",
  "complex-numbers",
  "connect",
  "crypto-square",
  "custom-set",
  "diamond",
  "diffie-hellman",
  "dominoes",
  "error-handling",
  "etl",
  "flatten-array",
  "food-chain",
  "forth",
  "go-counting",
  "grade-school",
  "grep",
  "hangman",
  "house",
  "isbn-verifier",
  "isogram",
  "kindergarten-garden",
  "knapsack",
  "largest-series-product",
  "linked-list",
  "list-ops",
  "luhn",
  "markdown",
  "matching-brackets",
  "matrix",
  "meetup",
  "minesweeper",
  "nth-prime",
  "nucleotide-count",
  "ocr-numbers",
  "palindrome-products",
  "parallel-letter-frequency",
  "pascals-triangle",
  "phone-number",
  "pig-latin",
  "poker",
  "prime-factors",
  "proverb",
  "pythagorean-triplet",
  "queen-attack",
  "rail-fence-cipher",
  "rational-numbers",
  "rectangles",
  "rest-api",
  "robot-name",
  "robot-simulator",
  "roman-numerals",
  "rotational-cipher",
  "run-length-encoding",
  "saddle-points",
  "satellite",
  "series",
  "sieve",
  "simple-cipher",
  "simple-linked-list",
  "spiral-matrix",
  "sublist",
  "sum-of-multiples",
  "tournament",
  "transpose",
  "tree-building",
  "triangle",
  "twelve-days",
  "two-bucket",
  "variable-length-quantity",
  "word-count",
  "word-search",
  "wordy",
  "yacht",
  "zebra-puzzle",
  "zipper"
]
</details>
```

There might be other exercises marked as easy that might not have stubs, so it might be worth to also double-check those.

About other tracks...I don't know the state of stubs in other tracks other than the Go track, which has them. Also note the inclusion of stubs is something that each track can decide independently, some tracks might decide to not include them on purpose. If you are thinking on contributing a similar change to other tracks, please confirm the stance of each track on including stubs before proceeding.

**Comment by jcovington16** [comment]
I will go through and knock out A-G just to give someone else an opportunity to knock out a few and it seems no one else jumped on this @andrerfcsantos.

**Comment by andrerfcsantos** [comment]
Cool! Go ahead!

**Comment by mvincentbb** [comment]
Please I'm new here. I want to contribute on this issue. Is possible

**Comment by mvincentbb** [comment]
Please, I’m new here. I want to contribute on this issue. Is it possible?

**Comment by andrerfcsantos** [comment]
@mvincentbb Yes! Feel free to help with this issue.

**Comment by gabbyhershey** [comment]
I want to take this on too

**Comment by andrerfcsantos** [comment]
@gabbyhershey Feel free to do so. Several people can work on this. If you are thinking on working on several exercises, it's probably a good idea to say your plan just to avoid repeated work.

**Comment by jennylia** [comment]
I'll do a small "linked-list" one to get started. I'm gonna ignore 
```/*

Since this exercise has a difficulty of > 4 it doesn't come
with any starter implementation.
This is so that you get to practice creating classes and methods
which is an important part of programming in Java.

Please remove this comment when submitting your solution.

*/
``` for now

**Comment by jennylia** [comment]
my PR:
https://github.com/exercism/java/pull/2183

**Comment by andrerfcsantos** [comment]
@jennylia Cool! If in the PR you mention the issue, it is automatically linked here. Avoids you having to make a comment just to link the issue.

**Comment by PakkuDon** [comment]
Hello there! 
I wasn't sure which exercises had been claimed but I'll have a PR up shortly to add stubs for exercises W - Z

**Comment by andrerfcsantos** [comment]
@PakkuDon I think they are all pretty much open :)

**Comment by Trympyrym** [comment]
added all missed stubs i've found

**Comment by saaagarsingh** [comment]
@andrerfcsantos hey just wanted to know if all the exercises have got the stubs or not. If the issue is still open to work then do let me know. I'll start working on it.

**Comment by andrerfcsantos** [comment]
@saaagarsingh Some still don't. Feel free to add stubs to exercises where you don't see them.

And thank you for your interest in this. I think this is something that will massively benefit the track.

**Comment by Harsh4902** [comment]
Hey, if this issue is still open than I want to contribute in it

**Comment by andrerfcsantos** [comment]
@Harsh4902 Yes, we're still accepting contributions for this!

**Comment by Bellandrre** [comment]
Hi, I've tried to add stubs for one exercise here -
https://github.com/exercism/java/pull/2303#issue-1693283680

**Comment by shantanugaware** [comment]
Hello, I'm new here, is the issue fixed or does it still needs attention?

**Comment by keiravillekode** [comment]
The following still lack stubs:

```txt
allergies
all-your-base
alphametics
anagram
atbash-cipher
bank-account
beer-song
bob
book-store
bowling
change
circular-buffer
clock
complex-numbers
crypto-square
custom-set
diffie-hellman
dominoes
flatten-array
food-chain
forth
go-counting
grade-school
grep
hangman
house
knapsack
matching-brackets
meetup
minesweeper
nucleotide-count
ocr-numbers
palindrome-products
parallel-letter-frequency
pascals-triangle
phone-number
pig-latin
poker
prime-factors
pythagorean-triplet
queen-attack
rail-fence-cipher
rectangles
rest-api
robot-name
robot-simulator
roman-numerals
run-length-encoding
```

I grepped for "Since this exercise has a difficulty of"

**Comment by mnyon-grandkru** [comment]
Hi there. I created a PR to help fill in stubs for the PigLatinTranslator, but the check may need approval. What would I need to do in order to get that approval

**Comment by SleeplessByte** [comment]
Fixed by #2325

**Comment by SleeplessByte** [comment]
(woops didn't mean to close it)

**Comment by SleeplessByte** [comment]
Partially fixed by #2319

**Comment by sanderploegsma** [comment]
I went through the practice exercises and found two more exercises without stubs, opened PRs to add them. Once they are merged I believe this issue can be closed as completed!
